crates/swc_ecma_parser/src/parser/class_and_fn.rs

```diff
@@ -1615,10 +1615,22 @@ pub(super) trait FnBodyParser<Body> {
     fn parse_fn_body_inner(&mut self, is_simple_parameter_list: bool) -> PResult<Body>;
 }
 fn has_use_strict(block: &BlockStmt) -> Option<Span> {
-    match block.stmts.iter().find(|stmt| stmt.is_use_strict()) {
-        Some(Stmt::Expr(ExprStmt { span, expr: _ })) => Some(*span),
-        _ => None,
+    for stmt in &block.stmts {
+        match stmt {
+            Stmt::Expr(s) => match &*s.expr {
+                Expr::Lit(Lit::Str(..)) => {
+                    if stmt.is_use_strict() {
```

Comment:
What are the differences?

Reply:
Directives should be at the start of the file, and this PR changes the parser to match only at the start of a file.

Reply:
We skip non-use-strict directives, because those can be also a directive

Reply:
Check this
https://github.com/swc-project/swc/blob/f7baf240ba15286c9e381f3b037d677ee7b1f63c/crates/swc_ecma_transforms_module/src/util.rs#L199-L210